server/coordinator/node_picker.go

```diff
@@ -4,53 +4,78 @@ package coordinator
 
 import (
 	"context"
-	"strconv"
 	"time"
 
 	"github.com/CeresDB/ceresmeta/server/cluster/metadata"
 	"github.com/CeresDB/ceresmeta/server/hash"
 	"github.com/CeresDB/ceresmeta/server/storage"
 	"github.com/pkg/errors"
+	"github.com/spaolacci/murmur3"
 	"go.uber.org/zap"
 )
 
 type NodePicker interface {
-	PickNode(ctx context.Context, shardID storage.ShardID, registerNodes []metadata.RegisteredNode) (metadata.RegisteredNode, error)
+	PickNode(ctx context.Context, shardID storage.ShardID, shardTotalNum uint32, registerNodes []metadata.RegisteredNode) (metadata.RegisteredNode, error)
 }
 
-type ConsistentHashNodePicker struct {
-	logger       *zap.Logger
-	hashReplicas int
+type UniformityConsistentHashNodePicker struct {
+	logger *zap.Logger
 }
 
-func NewConsistentHashNodePicker(logger *zap.Logger, hashReplicas int) NodePicker {
-	return &ConsistentHashNodePicker{hashReplicas: hashReplicas, logger: logger}
+func NewUniformityConsistentHashNodePicker(logger *zap.Logger) NodePicker {
```

Comment:
The schedule can be placed on the cluster, so that different clusters can support different scheduling algorithms.